src/ChunkSender.h

```diff
@@ -53,87 +56,73 @@ class cChunkSender:
 {
 	typedef cIsThread super;
 public:
-	cChunkSender(void);
+	cChunkSender(cWorld & a_World);
 	~cChunkSender();
 
 	enum eChunkPriority
 	{
-		E_CHUNK_PRIORITY_HIGH   = 0,
-		E_CHUNK_PRIORITY_MEDIUM = 1,
-		E_CHUNK_PRIORITY_LOW    = 2,
+		E_CHUNK_PRIORITY_LEGENDARY = 0,
+		E_CHUNK_PRIORITY_HIGH,
+		E_CHUNK_PRIORITY_MEDIUM,
+		E_CHUNK_PRIORITY_LOW,
+		
 	};
 	
-	bool Start(cWorld * a_World);
+	bool Start();
 	
 	void Stop(void);
 	
-	/// Notifies that a chunk has become ready and it should be sent to all its clients
-	void ChunkReady(int a_ChunkX, int a_ChunkZ);
-	
 	/// Queues a chunk to be sent to a specific client
 	void QueueSendChunkTo(int a_ChunkX, int a_ChunkZ, eChunkPriority a_Priority, cClientHandle * a_Client);
+	void QueueSendChunkTo(int a_ChunkX, int a_ChunkZ, eChunkPriority a_Priority, std::list<cClientHandle *> a_Client);
 	
 	/// Removes the a_Client from all waiting chunk send operations
 	void RemoveClient(cClientHandle * a_Client);
 	
 protected:
-
-	/// Used for sending chunks to specific clients
-	struct sSendChunk
+	
+	struct sChunkQueue
 	{
-		int m_ChunkX;
-		int m_ChunkZ;
-		cClientHandle * m_Client;
-		
-		sSendChunk(int a_ChunkX, int a_ChunkZ, cClientHandle * a_Client) :
-			m_ChunkX(a_ChunkX),
-			m_ChunkZ(a_ChunkZ),
-			m_Client(a_Client)
-		{
-		}
-		
-		bool operator ==(const sSendChunk & a_Other)
+		eChunkPriority m_Priority;
+		cChunkCoords m_Chunk;
+
+		bool operator <(const sChunkQueue & a_Other) const
 		{
-			return (
-				(a_Other.m_ChunkX == m_ChunkX) &&
-				(a_Other.m_ChunkZ == m_ChunkZ) &&
-				(a_Other.m_Client == m_Client)
-			);
+			/* The Standard Priority Queue sorts from biggest to smallest
+			return true here means you are smaller than the other object, and you get pushed down.
+
+			The priorities go from HIGH (0) to LOW (2), so a smaller priority should mean further up the list
+			∴ return true (affirm we're "smaller", and get pushed down) only if our priority is bigger than theirs (they're more urgent)
```

Comment:
Some platforms / IDEs / compilers don't support them. Not sure what the standard says about this.